File: Part 2 - Regression/Section 6 - Polynomial Regression/TN_polynomial_regression.R


dataset = read.csv('Position_Salaries.csv')


dataset=dataset[,2:3]


dataset1 = read.csv('Position_Salaries.csv')
dataset1 = dataset1[2:3]


lin_reg = lm(formula=Salary~., data=dataset)

dataset$Level2 = dataset$Level^2
dataset$Level3 = dataset$Level^3
dataset$Level4 = dataset$Level^4

poly_reg=lm(formula = Salary~., data=dataset)

install.packages('ggplot2')
library('ggplot2')

ggplot()+
  geom_point(aes(x=dataset$Level, y=dataset$Salary), color = 'Red')+
#  geom_line(aes(x=dataset$Level, y=predict(lin_reg), color = 'Blue'))+
  geom_line(aes(x=dataset$Level, y=predict(lin_reg, newdata=dataset$level)), colour = 'blue')+
  ggtitle('Level V Salary')+
  xlab('Level')+
  ylab('Salary')

ggplot()+
  geom_point(aes(x=dataset$Level, y=dataset$Salary), color = 'Red')+
  #  geom_line(aes(x=dataset$Level, y=predict(lin_reg), color = 'Blue'))+
  geom_line(aes(x=dataset$Level, y=predict(poly_reg, newdata=dataset)), colour = 'blue')+
  ggtitle('Level V Salary')+
  xlab('Level')+
  ylab('Salary')

x_grid = seq(min(dataset$Level),max(dataset$Level), 0.1)

newdf=data.frame(Level = x_grid,
                 Level2 = x_grid^2,
                 Level3 = x_grid^3,
                 Level4 = x_grid^4)

ggplot()+
  geom_point(aes(x=dataset$Level, y=dataset$Salary), color = 'Red')+
  #  geom_line(aes(x=dataset$Level, y=predict(lin_reg), color = 'Blue'))+
  geom_line(aes(x=x_grid, y=predict(poly_reg, newdata=newdf)), colour = 'blue')+
  ggtitle('Level V Salary')+
  xlab('Level')+
  ylab('Salary')

File: Part 2 - Regression/Section 6 - Polynomial Regression/TN_polynomial_regression.py
# -*- coding: utf-8 -*-
"""
Created on Mon Dec 10 11:40:28 2018

@author: Thineth
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Position_salaries.csv')

X= dataset.iloc[:,1:2].values
y = dataset.iloc[:,2].values



from sklearn.linear_model import LinearRegression

lin_reg=LinearRegression()
lin_reg.fit(X,y)

from sklearn.preprocessing import PolynomialFeatures

poly_reg = PolynomialFeatures(degree=4)
X_poly=poly_reg.fit_transform(X)
#poly_reg.fit(X_poly,y)

lin_reg2=LinearRegression()
lin_reg2.fit(X_poly,y)

plt.scatter(X,y,color='red')
plt.plot(X,lin_reg.predict(X), color='blue')
plt.xlabel("Level")
plt.ylabel("Salary")
plt.title("Level to salary")
plt.show()

plt.scatter(X,y,color='red')
plt.plot(X, lin_reg2.predict(poly_reg.fit_transform(X)))

plt.show()

X_grid = np.arange(min(X), max(X),0.1)
X_grid2 = X_grid.reshape(len(X_grid), 1)




